pennylane/_device.py

```diff
@@ -145,9 +145,13 @@ def execute(self, queue, observables, parameters={}):
         Args:
             queue (Iterable[~.operation.Operation]): operations to execute on the device
             observables (Iterable[~.operation.Observable]): observables to measure and return
-            parameters (dict[int->list[ParameterDependency]]): Mapping from free parameter index to the list of
+            parameters (dict[int, list[ParameterDependency]]): Mapping from free parameter index to the list of
                 :class:`Operations <pennylane.operation.Operation>` (in the queue) that depend on it.
 
+        Keyword Args:
+            return_native_type (bool): If True, return the result in whatever type the device uses
+                internally, otherwise convert it into array[float]. Default: False.
+
```

Comment:
Maybe this kwarg should also be present in `QubitDevice`

Reply:
Yeah, about that. `QubitDevice` inherits `Device`, but it changes the signature of the `execute` method, which isn't usually a good practice. I was going to suggest to fix this if possible, but in another PR.

Reply:
@smite this is unfortunate, and due to the fact that we are still in the process of porting all the plugins to QubitDevice. Once the migration is over, the `Device.execute` method will be removed, and only defined in `QubitDevice`.

Reply:
Currently, default.qubit, Forest, and Cirq have been ported, while SF, default.tensor, Qiskit remain with the old device class.

Reply:
What's the reason why qubit devices and CV devices need to have a different signature for `execute`?

Reply:
They won't --- the signature for `execute` is the new plugin API, and will be used in both `QubitDevice` and `CVDevice`. The old `Device.execute` has been deprecated, however we can't fully remove it until all plugins have been ported.

Reply:
If both QubitDevice and CVDevice inherit Device and they both have the same interface for `execute`, wouldn't it then make sense to keep `execute` an abstract method of Device?

Reply:
Yep that's the plan :+1: